src/Riok.Mapperly/Descriptors/MappingBodyBuilders/MemberMappingBuilder.cs

```diff
@@ -84,6 +85,22 @@ public static bool TryBuild(
             return false;
         }
 
+        var memberTargetNullable = memberMappingInfo.TargetMember.MemberType.IsNullable();
+        var delegateTargetNullable = delegateMapping.TargetType.IsNullable();
+        var memberSourceNullable = memberMappingInfo.IsSourceNullable;
+        var delegateSourceNullable = delegateMapping.SourceType.IsNullable();
+
+        if (memberSourceNullable && !memberTargetNullable && !(delegateSourceNullable && !delegateTargetNullable))
```

Comment:
Shouldn't the first two and the second two have an OR relationship instead of AND?

Reply:
The intention here is that the last case `delegateSourceNullable && !delegateTargetNullable` happens when the delegate function is manually defined and already has null handling, in which case we don't want to give the diagnostic. For example, test `ShouldPassNullValueToNullableUserMappingMethod` uses 

```
private string MapString(string? source)
    => source ?? "(null)";
```

Which would still give a diagnostic without that last check

Reply:
But `ShouldPassNullValueToNullableUserMappingMethod ` should diagnostic for the member Value2, but not for the member Value.

Reply:
It does right now, looking at the ShouldPassNullValueToNullableUserMappingMethod verified file